src/main/java/com/google/api/codegen/transformer/php/PhpPackageMetadataNamer.java

```diff
@@ -21,12 +21,40 @@
 /** PHPPackageMetadataNamer provides PHP specific names for metadata views. */
 public class PhpPackageMetadataNamer extends PackageMetadataNamer {
   private Name serviceName;
-  private String domainLayerLocation;
+  private String metadataIdenfifier;
 
   public PhpPackageMetadataNamer(String packageName, String domainLayerLocation) {
-    // Get the service name from the package name by removing the version suffix (if any).
-    this.serviceName = getApiNameFromPackageName(packageName);
-    this.domainLayerLocation = domainLayerLocation;
+    String packageNameWithoutVersion = PhpPackageUtil.getPackageNameBeforeVersion(packageName);
+    String[] packagePieces = PhpPackageUtil.splitPackageName(packageNameWithoutVersion);
+
+    this.serviceName = Name.upperCamel(packagePieces);
+
+    // To build metadata identifier, update pieces to be lowercase, so each piece is treated as a
+    // single piece in the Name object
+    for (int i = 0; i < packagePieces.length; i++) {
+      packagePieces[i] = packagePieces[i].toLowerCase();
```

Comment:
It will have the same effect here, but will slightly change the value in `serviceName`, which interprets the camel casing in the original. Rather than iterate here, we could perform two splits, one on `basePackageName` for `serviceName`, and another of `basePackageName.toLowerCase()` for `metadataIndentifier` - but I'm not sure if that is preferable.